domain/ide/workspace/folder_test.go

```diff
@@ -507,6 +507,45 @@ func Test_processResults_ShouldNotSendAnalyticsToAPIIfDisabled(t *testing.T) {
 	f.processResults(data)
 }
 
+func Test_processResults_ShouldCountSeverityByProduct(t *testing.T) {
+	c := testutil.UnitTest(t)
+	c.SetAnalyticsEnabled(false)
+
+	engineMock, gafConfig := setUpEngineMock(t, c)
+
+	f, _ := NewMockFolderWithScanNotifier(notification.NewNotifier())
+	const filePath = "path1"
+
+	scanData1 := snyk.ScanData{
+		Product: product.ProductOpenSource,
+		Issues: []snyk.Issue{
+			{Severity: snyk.Critical, Product: product.ProductOpenSource},
+			{Severity: snyk.Critical, Product: product.ProductOpenSource},
+			{Severity: snyk.High, Product: product.ProductOpenSource},
+		},
+	}
+
+	// scanData2 := snyk.ScanData{
+	// 	Product: product.ProductInfrastructureAsCode,
+	// 	Issues: []snyk.Issue{
+	// 		{Severity: snyk.Low, Product: product.ProductInfrastructureAsCode},
+	// 		{Severity: snyk.High, Product: product.ProductInfrastructureAsCode},
+	// 	},
+	// }
+
+	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
+	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).Times(0)
+
+	// Act
+	f.processResults(scanData1)
+	// f.processResults(scanData2)
+
+	// Assert
+	if got, want := scanData1.SeverityCount[product.ProductOpenSource][snyk.Critical.String()], 2; got != want {
+		t.Errorf("got %v, want %v", got, want)
+	}
```

Comment:
WIP: I'm not sure how to assert `scanData`... 
Change `processResults` to accept `scanData` as a pointer? 🤔

Reply:
suggestion: how about `require.Equal(t, 2, scanData1.SeverityCount[product.ProductOpenSource][snyk.Critical.String()])`? Would be easier for me to understand. Not sure about your question, calling :)